main.go

```diff
@@ -622,13 +618,19 @@ func transformCompile(args []string) ([]string, error) {
 	}
 
 	newPaths := make([]string, 0, len(files))
-	for i, file := range files {
+
+	for _, file := range files {
+		// Process directives before obfuscating any names of the package,
+		// this is needed for functions which are linknamed to another package,
+		// but also used in the package declaring them.
```

Comment:
I don't understand. Can you provide an example of a case that would break without the separation of the original loop?

Reply:
I found this while poking at the runtime.

For example the time_now call can be obfuscated here:

https://go.googlesource.com/go/+/go1.16.2/src/runtime/mgc.go#1703

before finding this linkname directive along with the declaration in a separate file

https://go.googlesource.com/go/+/go1.16.2/src/runtime/timestub.go#14

This leads to the call being obfuscated and the function declaration being blacklisted later because of the linkname directive.

Reply:
I see, good point. Could you also add a test case for it? The files being compiled should be in alphanumeric order, so you could have `a.go` call a linkname function, and `b.go` declare the linkname function.

I get that std covers this as a regression test right now, but it won't run as part of `go test -short`, and maybe the runtime package will be refactored in the future to no longer act as a regression test.

I bet you could also split this up into a smaller PR, which we could approve and merge today, without #313.